Week3/hanoi/src/Hanoi.java

```diff
@@ -1,70 +1,54 @@
-import java.util.Stack;
-
+// Set bars to execute hanoi algorithm.
 public class Hanoi {
-  int num_disks;
-
-  Stack<Integer> bar1 = new Stack<>();
 
-  Stack<Integer> bar2 = new Stack<>();
+  int numDisks = 0;
 
-  Stack<Integer> bar3 = new Stack<>();
+  // Initialize the numDisks and put disks in bar1 from larger disks to smaller disks.
+  // Numbers in the bar mean the size of a disk.
+  public Hanoi(int numDisks) {
+    this.numDisks = numDisks;
+  }
 
-  public Hanoi (int num_disks) {
-    this.num_disks = num_disks;
-    
-    for (int i=num_disks; i>=1; i--) {
-      bar1.push(i);
-    }
+  // Print log of disk movement.
+  private void moveDisk(int diskNumber, int fromBar, int toBar) {
+    System.out.println("Disk "+diskNumber+" moved from bar "+fromBar+" to bar "+toBar);
   }
 
-  public int hanoi (int num_disks, int from_bar, int second_bar, int to_bar) {
-    if (num_disks == 1) {
+  // Returns the answer of Hanoi problem with initialized numDisks
+  public int calculateHanoiSteps() throws HanoiInputException {
+    return calculateHanoiSteps(numDisks, 1, 2, 3);
+  }
 
-      /* Pop a disk from from_bar */
-      int disk_toMove = 0;
+  // Returns the answer of Hanoi problem with given numDisks.
+  // Prints disk movement log.
+  public int calculateHanoiSteps(int numDisks, int fromBar, 
+                                 int secondBar, int toBar) throws HanoiInputException {
+    // The number of disks should be a positive integer.
+    if (numDisks < 1) {
+      throw new HanoiInputException("The number of disks should be a positive number.");
+    }
 
-      switch (from_bar) {
-        case 1: 
-          disk_toMove = bar1.pop();
-          break;
-        case 2:
-          disk_toMove = bar2.pop();
-          break;
-        case 3:
-          disk_toMove = bar3.pop();
-          break;
-        default:
-          assert(false);
-          break;
-      }
-  
-      /* Print disk movement */
-      System.out.println("Disk "+disk_toMove+" moved from bar "+from_bar+" to bar "+to_bar);
+    // from bar, second bar and to bar numbers should be beteen 0 and 4, exclusively.
+    if ((fromBar < 1) || (secondBar < 1) || (toBar < 1) 
+        || (fromBar > 3) || (secondBar > 3) || (toBar > 3)) {
```

Comment:
I suggest indenting as below

 if ((fromBar < 1) || (secondBar < 1) || (toBar < 1) || 
     (fromBar > 3) || (secondBar > 3) || (toBar > 3)) {

because it's more straightforward to look at what conditions are exist